src/setfit/modeling.py

```diff
@@ -132,16 +132,17 @@ def forward(
 
         return outputs
 
-    def predict_proba(self, x_test: torch.Tensor) -> torch.Tensor:
+    def predict_proba(self, x_test: Union[torch.Tensor, "ndarray"]) -> torch.Tensor:
         self.eval()
 
-        return self(x_test)
-
-    def predict(self, x_test: Union[torch.Tensor, "ndarray"]) -> Union[torch.Tensor, "ndarray"]:
         is_tensor = isinstance(x_test, torch.Tensor)
         if not is_tensor:  # then assume it's ndarray
             x_test = torch.Tensor(x_test).to(self.device)
 
+        return self(x_test)
+
+    def predict(self, x_test: Union[torch.Tensor, "ndarray"]) -> Union[torch.Tensor, "ndarray"]:
+        is_tensor = isinstance(x_test, torch.Tensor)
         probs = self.predict_proba(x_test)
 
         if self.out_features == 1:
```

Comment:
If the above suggestions sounds good, then since we converted `x_test` in `predict_proba`, we could use `is_tensor` to determine using `numpy` or `torch` for line:149 and line:151. And we don't need line:153-154 anymore.

This suggestion will create nested if statements since it needs to choose using `numpy` or `torch` in line:149 and line:151. But in PR #203, I treat binary classification as 2-class classification using `CrossEntropyLoss`, so the nested if statements will disappear if the PR is accepted. 

If you like this idea, feel free make the change first and I will update PR #203 after yours is merged :)